cvmfs/catalog_mgr.h

```diff
@@ -115,16 +115,18 @@ class InodeGenerationAnnotation : public InodeAnnotation {
   uint64_t inode_offset_;
 };
 
-
+template <class CatalogT>
 class AbstractCatalogManager;
+
 /**
  * Here, the Cwd Buffer is registered in order to save the inodes of
  * processes' cwd before a new catalog snapshot is applied
  */
+template <class CatalogT>
 class RemountListener {
```

Comment:
Actually I think that we don't use the `RemountListener` anywhere anymore.  Can you try to remove it, including the calling code in `AbstractCatalogManager`?